Zend/zend_operators.c

```diff
@@ -54,7 +57,30 @@ static _locale_t current_locale = NULL;
 
 #define TYPE_PAIR(t1,t2) (((t1) << 4) | (t2))
 
-#if __SSE2__
+#ifdef ZEND_INTRIN_AVX2_NATIVE
+#define HAVE_BLOCKCONV
+
+#define BLOCKCONV_INIT_RANGE(start, end) \
+	const __m256i blconv_offset = _mm256_set1_epi8((signed char)(SCHAR_MIN - start)); \
+	const __m256i blconv_threshold = _mm256_set1_epi8(SCHAR_MIN + (end - start) + 1);
+
+#define BLOCKCONV_STRIDE sizeof(__m256i)
+
+#define BLOCKCONV_INIT_DELTA(delta) \
+	const __m256i blconv_delta = _mm256_set1_epi8(delta);
+
+#define BLOCKCONV_LOAD(input) \
+	__m256i blconv_operand = _mm256_loadu_si256((__m256i*)(input)); \
+	__m256i blconv_mask = _mm256_cmpgt_epi8(blconv_threshold, _mm256_add_epi8(blconv_operand, blconv_offset));
```

Comment:
Nit: Maybe this should be changed to lt instead of gt with the argument swapped in order to be symmetric with the SSE2 version.

Reply:
:smile:  There is no such thing as `_mm256_cmplt_epi8`. That's why I had to change to `cmpgt` and swap the arguments to get the code to compile.

Ref: https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_mm256_cmplt_epi8

If we want to make it symmetric, then we would need to change the SSE2 version.